File: microservices/user-service/src/main/java/com/halcyon/tinder/userservice/dto/error/ValidationErrorsResponse.java
package com.halcyon.tinder.userservice.dto.error;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.time.Instant;
import java.util.List;
import java.util.Map;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;

@Builder
@Data
@AllArgsConstructor
public class ValidationErrorsResponse {

    @JsonProperty("timestamp")
    private Instant timestamp;

    @JsonProperty("status")
    private Integer status;

    @JsonProperty("errors")
    private List<Map<String, String>> errors;
}

File: common/exception-core/src/main/kotlin/com/halcyon/tinder/exceptioncore/ValidationErrorsResponse.kt
package com.halcyon.tinder.exceptioncore

import com.fasterxml.jackson.annotation.JsonProperty
import lombok.Builder
import java.time.Instant

@Builder
data class ValidationErrorsResponse (
    @JsonProperty("timestamp")
    val timestamp: Instant = Instant.now(),

    @JsonProperty("status")
    val status: Int,

    @JsonProperty("errors")
    val errors: List<Map<String, String>>
)